src/lib/mongoBackend/mongoRegistrationsGet.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+*
+* Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <vector>
+
+#include "mongo/client/dbclient.h"
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "common/sem.h"
+#include "common/statistics.h"
+#include "common/errorMessages.h"
+#include "rest/OrionError.h"
+#include "rest/HttpStatusCode.h"
+#include "apiTypesV2/Registration.h"
+#include "mongoBackend/dbConstants.h"
+#include "mongoBackend/safeMongo.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "mongoBackend/connectionOperations.h"
+#include "mongoBackend/MongoCommonRegister.h"
+#include "mongoBackend/mongoRegistrationsGet.h"
+
+
+
+/* ****************************************************************************
+*
+* mongoRegistrationsGet - 
+*/
+void mongoRegistrationsGet
+(
+  std::vector<ngsiv2::Registration>*  regV,
+  const std::string&                  tenant,
+  const std::vector<std::string>&     servicePathV,
+  OrionError*                         oeP
+)
+{
+  bool         reqSemTaken = false;
+  std::string  err;
+  mongo::OID   oid;
+  StatusCode   sc;
+  int          limit  = 5000;  // FIXME: Pagination: limit, offset and count are a preparation for it - see mongoGetSubscriptions()
+  int          offset = 0;
+  long long    count;
+
+  reqSemTake(__FUNCTION__, "Mongo Get Registrations", SemReadOp, &reqSemTaken);
+
+  LM_T(LmtMongo, ("Mongo Get Registrations"));
+
+  std::auto_ptr<mongo::DBClientCursor>  cursor;
+  mongo::Query                          q;
+
+  if ((servicePathV.size() != 0) && (servicePathV[0] != "/#"))
+  {
+    q = BSON(REG_SERVICE_PATH << fillQueryServicePath(servicePathV));
```

Comment:
This is not correct. Exact match is needed:

> Fiware-ServicePath header is taken into account in `GET /v2/subscriptions` and `GET /v2/registrations` in order to narrow down the results to subscriptions/registrations that use *exactly* that service path as query.

Reply:
Already responded, issue to ne opened for this discussion

Reply:
I think we already have an issue for this: telefonicaid#3051. See #17 (comment).

Reply:
Issue telefonicaid#3051 is about GET /v2/registration/XXX, where service path isn't really interesting as the exact registration is already identified by its ID.

Let's just decide something here so we can close the PR.

The easy way is to make an exact match with `servicePathV[0]`
At the same time we could add an error if `servicePathV.size() > 1` ...

Because exact match with a vector ...

Reply:
Exact match with `servicePathV[0]` implemented in 108dde4

Reply:
I will comment on this, but in a new "fresh" comment thread, on the new version of the code.